### PR TITLE
fix: POIs open times issue

### PR DESCRIPTION
**Pull request by ardasnturk**
- fixed an issue with POIs showing as open if the day is not selected on days when POIs are closed

SVA-925

## How to test:
the cause of the problem:
some POIs do not have the day and time selected. if the day and time are not selected, the code automatically thinks that the POI is open because it does not check the day.

- go to POI with id `38` in CMS
- make the `geöffnet` of the current day false
- please make sure you select the correct day via the CMS when reviewing
- in the app go to `Touren und Orte` -> `Gastronomie` -> `"Panda Bistro"`
- check whether the selected POI is open or closed

## Screenshots:

image

|before|after|
|--|--|
Simulator Screen Shot - iPhone 14 Pro Max - 2023-03-09 at 10 07 47 | image

### CODE REVIEW

File: src/helpers/openingHoursHelper.ts [hunk 1]
```diff
@@ -236,7 +236,7 @@ export const isOpen = (
       .filter((info) => info.weekday === getReadableDay(now) || !info.weekday)
       .filter((info) => isOpeningTimeForDate(info, now));
 
-    const todaysOpenTimes = todaysTimes.filter((info) => !!info.open);
+    const todaysOpenTimes = todaysTimes.filter((info) => !!info.open && !!info.weekday);
```
Comment (donni106) [inline src/helpers/openingHoursHelper.ts:239]: `info.weekday` is this field, or am i wrong?
image

Comment (ardasnturk) [reply src/helpers/openingHoursHelper.ts:239]: Yes, you are right. Unfortunately, it's not a method we can always trust. But we should definitely check the `weekday`.

Comment (donni106) [reply src/helpers/openingHoursHelper.ts]: reworked with 55e3e18, please have a look

**Review by ardasnturk — COMMENTED** [review]
I have tried different methods and I can say that in general it works well. 👍 🎉

File: src/helpers/openingHoursHelper.ts [hunk 2]
```diff
@@ -229,12 +230,15 @@ export const isOpen = (
...
-      .filter((info) => isOpeningTimeForDate(info, now));
+    // filter all times that
+    //  - are corresponding to the correct weekday or are without a weekday but with a time value
+    //  - have a currently valid date range
+    const todaysTimes = openingHours.filter(
+      (info) =>
+        (info.weekday === getReadableDay(now) ||
+          (!info.weekday && (info.timeFrom || info.timeTo))) &&
+        isOpeningTimeForDate(info, now)
+    );
```
[the remark this answers is not in the document]
Comment (donni106) [reply src/helpers/openingHoursHelper.ts:241]: good question. i have added boolean checks to ensure its presence: `!!info.timeFrom || !!info.timeTo`

File: src/helpers/calendarHelper.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { LocaleConfig } from 'react-native-calendars';
 
 export const setupLocales = () => {
+  LocaleConfig.defaultLocale = 'de';
```
[the remark this answers is not in the document]
Comment (donni106) [reply src/helpers/calendarHelper.ts:4]: i found this line at the bottom and i din't expected something to come after the locale strings translations.
this way we cannot oversee the default locale setting

File: src/components/survey/SurveyAnswer.tsx [hunk 4]
```diff
@@ -2,7 +2,8 @@ import React, { SetStateAction, useCallback } from 'react';
 import { StyleSheet, View } from 'react-native';
 
 import { colors, normalize } from '../../config';
-import { getAnswerLabel, imageWidth } from '../../helpers';
+import { getAnswerLabel } from '../../helpers';
+import { imageWidth } from '../../helpers/imageHelper';
```
[the remark this answers is not in the document]
Comment (donni106) [reply src/components/survey/SurveyAnswer.tsx:6]: jest crashed with an error message that these functions seems to be not defined. this was very confusing. with changing the imports it worked, which i was trying to describe in the commit message with

> - imports needed to be readjusted and more explicit

**Review by donni106 — APPROVED** [review]

File: src/helpers/openingHoursHelper.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/helpers/openingHoursHelper.ts]: https://github.com/smart-village-solutions/smart-village-app-app/pull/768/commits/f34dfb1c5ad194282b6f1e8619fb9dc7d95f8441

**Pull request merged**